### PR TITLE
Adding population variable on the Census Tract level

### PR DESCRIPTION
**Pull request by Niklewa**
Adding a population variable and creating a source of truth for Census Tract variables.

### CODE REVIEW

**Comment by Niklewa** [comment]
In this PR, I have downloaded Census API data about the population from 2010 to 2022. Given the complexity of Census tracts, the dataset was split into pre- and post-2020 periods, reflecting changes in Census tract definitions.

Additionally, a new `VariableCleanerCT` class and `CTDataGrabberCSV` were developed for use with this data level. I have also created tests to verify the data's accuracy.

Interestingly, a linting problem arises during Git checks, my linting does not indicate any necessary changes.

**Comment by Niklewa** [comment]
I fixed the lint issue, I updated `black` library on my setup and it made the necessary changes.

**Review by rfl-urbaniak — CHANGES_REQUESTED** [review]
Do I understand correctly that the raw data import logic now lives only in a notebook which is called "clean_population_tracts"? 

If yes, this might make future updates hard, no one will rember what's what and user will be forced to browse through experimental notebooks to find the right chunks of code. 

Please extract the core of the data import code, put in a script, and place in new folder `cities/utils/scraping`.

Do I see correctly that you did data grabber for csv but for now did not do one for DB, postponing this work potentially for later?

File: cities/utils/data_grabber.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ def find_repo_root() -> Path:
 
 def check_if_tensed(df):
     years_to_check = ["2015", "2018", "2019", "2020"]
-    check = df.columns[2:].isin(years_to_check).any().any()
```
[the remark this answers is not in the document]
Comment (Niklewa) [reply cities/utils/data_grabber.py:20]: I have changed that during some debugging, the first version seems to be better, as it omits two first columns.

File: cities/utils/data_grabber.py [hunk 2]
```diff
@@ -68,31 +68,104 @@ def __init__(self):
         sys.path.insert(0, self.data_path)
 
 
-def list_available_features(level="county"):
+class CTDataGrabberCSV(DataGrabberCSV):
+    def __init__(
+        self, level_DG: str = "pre_2020"
```
[the remark this answers is not in the document]
Comment (Niklewa) [reply cities/utils/data_grabber.py:73]: Instead I used `ct_time_period`

File: cities/utils/data_grabber.py [hunk 3]
```diff
@@ -68,31 +68,104 @@ def __init__(self):
...
     if level == "county":
         folder_path = f"{root}/data/processed"
     elif level == "msa":
         folder_path = f"{root}/data/MSA_level"
+    elif level == "census_tract":
+        folder_path = f"{root}/data/Census_tract_level"
     else:
-        raise ValueError("Invalid level. Please choose 'county' or 'msa'.")
+        raise ValueError(
+            "Invalid level. Please choose 'county', 'census_tract' or 'msa'."
```
[the remark this answers is not in the document]
Comment (Niklewa) [reply cities/utils/data_grabber.py:138]: resolved with changing the name of `level_DG`

File: cities/utils/clean_variable.py [hunk 4]
```diff
@@ -206,3 +206,45 @@ def communities_tracts_to_counties(
...
+
+class VariableCleanerCT(
+    VariableCleanerMSA
+):  # this class inherits functionalites of two previous classes, but works at the Census Tract level
+    def __init__(
+        self,
+        variable_name: str,
+        path_to_raw_csv: str,
+        time_interval: str,  # pre2020 or post2020
+        year_or_category: str = "Year",
```
[the remark this answers is not in the document]
Comment (rfl-urbaniak) [reply cities/utils/clean_variable.py:219]: still, the name is confusing. Please change.:)

File: cities/utils/data_grabber.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rfl-urbaniak) [reply cities/utils/data_grabber.py]: which version do you call the first version, the one that you deleted?

**Review by rfl-urbaniak — CHANGES_REQUESTED** [review]
two really small things left; also, the setup now contains the new dependency

File: cities/utils/data_grabber.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Niklewa) [reply cities/utils/data_grabber.py]: I returned to the original version, which I called the first version

File: cities/utils/clean_variable.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Niklewa) [reply cities/utils/clean_variable.py:219]: Changed to `year_or_category_column_label`

**Review by rfl-urbaniak — APPROVED** [review]

**Pull request merged**